operations/mimir-mixin/dashboards/tenants.libsonnet

```diff
@@ -530,5 +530,106 @@ local filename = 'mimir-tenants.json';
         ) +
         { legend: { show: false } },
       )
-    ),
+    )
+
+    .addRow(
+      $.row('Read Path - Query Frontend Insights')
+      .addPanel(
+        local title = 'Rate of Read Requests - Query Frontend';
+        $.panel(title) +
+        $.queryPanel(
+          'sum(rate(cortex_query_frontend_queries_total{%s, container="query-frontend", user="$user"}[$__rate_interval]))' % $.namespaceMatcher(),
+          'rate'
+        )
+      )
+      .addPanel(
+        local title = 'Rate of Read Requests - Ruler Query Frontend';
+        $.panel(title) +
+        $.queryPanel(
+          'sum(rate(cortex_query_frontend_queries_total{%s, container="ruler-query-frontend", user="$user"}[$__rate_interval]))' % $.namespaceMatcher(),
+          'rate'
+        )
+      )
+    )
+    .addRow(
+      $.row('Read Path - Query Scheduler Insights')
+      .addPanel(
+        local title = 'Number of Queries Queued - Query Scheduler';
+        $.panel(title) +
+        $.queryPanel(
+          [
+            'sum(cortex_query_scheduler_queue_length{%s, container="query-scheduler", user="$user"})' % $.namespaceMatcher(),
+          ],
+          [
+            'queue length',
+          ],
+        )
+      )
+      .addPanel(
+        local title = 'Number of Queries Queued - Ruler Query Scheduler';
+        $.panel(title) +
+        $.queryPanel(
+          [
+            'sum(cortex_query_scheduler_queue_length{%s, container="ruler-query-scheduler", user="$user"})' % $.namespaceMatcher(),
+          ],
+          [
+            'queue length',
+          ],
+        )
+      )
+    )
+
+    .addRow(
+      ($.row('Read Path - Queries') + { collapse: true })
+      .addPanel(
+        local title = 'Query Success vs Failure Rate';
+        $.panel(title) +
+        $.queryPanel(
+          [
+            'sum(rate({%s, name=~"query-frontend.*"} |= "msg=\\"query stats\\"" |= "user=$user" |= "status=success" [$__interval]))' % $.namespaceMatcher(),
```

Comment:
Why do we need to query Loki? Can't we just look at the query-frontend HTTP response code?